py/torch_tensorrt/dynamo/conversion/_TRTInterpreter.py

```diff
@@ -629,35 +657,68 @@ def run(
         assert serialized_engine
 
         _LOGGER.info(
-            f"Build TRT engine elapsed time: {datetime.now() - build_engine_start_time}"
+            f"Build weight-stripped TRT engine elapsed time: {datetime.now() - build_engine_start_time}"
         )
         _LOGGER.info(f"TRT Engine uses: {serialized_engine.nbytes} bytes of Memory")
 
         self._save_timing_cache(
             builder_config, self.compilation_settings.timing_cache_path
         )
-        if (
-            self.engine_cache is not None
-            and self.compilation_settings.cache_built_engines
-        ):
-            self.engine_cache.insert(
-                hash_val,
-                (
-                    serialized_engine,
-                    self._input_names,
-                    self._output_names,
-                    self.input_specs,
-                    self.compilation_settings,
-                    self.weight_name_map,
-                ),
-            )
+
+        # refittable engine
+        if not self.compilation_settings.immutable_weights:
+            # Engine caching only for refittable engine
+            if (
+                self.engine_cache is not None
+                and self.compilation_settings.cache_built_engines
+            ):
+                # Cache the weight-stripped engine
+                if self.compilation_settings.strip_engine_weights:
+                    weight_stripped_serialized_engine = serialized_engine
+                else:
+                    # Serialize the refitted engine where the EXCLUDE_WEIGHTS flag must be cleared
+                    runtime = trt.Runtime(TRT_LOGGER)
```

Comment:
Why do we refit then strip the weights again? If refit is enabled shouldnt the builder always us a weight stripped engine?